### PR TITLE
adding github action for releasing it to npm

### PR DESCRIPTION
**Pull request by vdua**
## Description
Added an action so that the package can be released to npm repository. 
Added some documentation and infra for generating jsdocs as well. 
Removed antlr from dependencies and moved to devDependencies.

## Tasks

- [ ] I have signed the [CLA](http://adobe.github.io/cla.html).
- [ ] I have written tests and verified that they fail without my change.

### CODE REVIEW

**Comment by vdua** [comment]
The workflow action has been taken from https://github.com/adobe/starter-repo/blob/master/.github/workflows/npm-publish.yml

File: package.json [hunk 1]
```diff
@@ -38,15 +36,19 @@
     "@babel/plugin-proposal-class-properties": "^7.16.7",
     "@babel/plugin-transform-modules-commonjs": "^7.16.8",
     "@babel/preset-env": "^7.16.11",
+    "antlr4": "^4.9.2",
```
Comment (JohnBrinkman) [inline package.json:39]: When we install "^4.9.2" of antlr, we end up with version 4.11.0
When I tried this, the antlr-based code generation / runtime failed -- they changed the way the modules are built in later versions of antlr.
Did you try doing a fresh `npm install` and code generation?

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  npm-publish:
+    name: npm-publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@master
+      - name: Set up Node.js
+        uses: actions/setup-node@master
+        with:
+          node-version: 10.0.0
```
Comment (JohnBrinkman) [inline .github/workflows/npm-publish.yml:16]: Not sure how important node-version is ... but this is a very old version.  Could we choose node 17?

File: package.json (hunk 1 above)
Comment (vdua) [reply package.json]: that change was unintentional. Reverted it.

**Pull request merged**